src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDelegateCreationExpression.cs

```diff
@@ -0,0 +1,1418 @@
+                ReturnedValue: ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1) (Syntax: '1')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS8030: Anonymous function converted to a void returning delegate cannot return a value
+                //         /*<bind>*/Action a = delegate() { return 1; };/*</bind>*/
+                Diagnostic(ErrorCode.ERR_RetNoObjectRequiredLambda, "return").WithLocation(7, 43)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalDeclarationStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DelegateCreationExpression_DelegateExpression_InvalidArgumentType()
+        {
+            string source = @"
+using System;
+class Program
+{
+    void Main()
+    {
+        /*<bind>*/Action a = delegate(int i) { };/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+IVariableDeclarationStatement (1 declarations) (OperationKind.VariableDeclarationStatement, IsInvalid) (Syntax: 'Action a =  ... int i) { };')
+  IVariableDeclaration (1 variables) (OperationKind.VariableDeclaration, IsInvalid) (Syntax: 'Action a =  ... int i) { };')
+    Variables: Local_1: System.Action a
+    Initializer: IDelegateCreationExpression (OperationKind.DelegateCreationExpression, Type: System.Action, IsInvalid) (Syntax: 'delegate(int i) { }')
+        Target: IAnonymousFunctionExpression (Symbol: lambda expression) (OperationKind.AnonymousFunctionExpression, Type: null, IsInvalid) (Syntax: 'delegate(int i) { }')
+            IBlockStatement (0 statements) (OperationKind.BlockStatement, IsInvalid) (Syntax: '{ }')
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS1593: Delegate 'Action' does not take 1 arguments
+                //         /*<bind>*/Action a = delegate(int i) { };/*</bind>*/
+                Diagnostic(ErrorCode.ERR_BadDelArgCount, "delegate(int i) { }").WithArguments("System.Action", "1").WithLocation(7, 30)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalDeclarationStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void DelegateCreationExpression_ImplicitMethodBinding()
+        {
+            string source = @"
+using System;
+class Program
+{
+    void Main()
+    {
+        Action a = /*<bind>*/M1/*</bind>*/;
+    }
+    void M1() { }
+}
+";
+
+            // Currently, GetOperation returns a straight BoundMethodGroup for the IdentifierNameSyntax. We can't directly
+            // convert a BoundMethodGroup to an IDelegateCreationExpression, so this returns OperationKind.None. When Heejae's
+            // rewrite of GetOperation is complete, this should return the IDelegateCreationExpression.
```

Comment:
>rewrite of GetOperation is complete, this should return the IDelegateCreationExpression. [](start = 15, length = 88)

This is probably incorrect because IDelegateCreationExpression is supposed to be marked as Implicit and getting IOperation for a syntax node isn't supposed to return implicit nodes.

Reply:
This is correct. I have separate tests to verify just the method binding, as well as the whole tree here.

---
In reply to: [140102041](https://github.com/dotnet/roslyn/pull/22056#discussion_r140102041) [](ancestors = 140102041)